### PR TITLE
[16.0][IMP] base_global_discount: Add check global discount in product

### PR DESCRIPTION
**Pull request by Rferri44-S73**
Add check global discount in products

### CODE REVIEW

**Review by QuocDuong1306 — APPROVED** [review]
lgtm

**Review by cyrilmanuel — APPROVED** [review]
LGTM

**Comment by rafaelbn** [comment]
Hello! :-)

This module is base_global_discount and not base_config_discount

This should be reviewed by @ferran-S73 @rousseldenis or @pedrobaeza 

Thank you!

**Comment by ferran-S73** [comment]
Hi @rafaelbn since the original owner of the PR doesn't work here anymore I fixed the commit name myself. As for the changes in here LGTM

**Comment by ferran-S73** [comment]
I guess the title of the PR has to be changed by a maintainer

File: base_global_discount/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 {
     "name": "Base Global Discount",
-    "version": "16.0.1.0.0",
+    "version": "16.0.1.1.0",
```
Comment (simahawk) [inline base_global_discount/__manifest__.py]: module version will be changed by the bot

**Review by simahawk — CHANGES_REQUESTED** [review]
LG overall, just a quick code review tho.

TODO:

1. rollback module version
2. remove odoo version from commit

@pedrobaeza anyone from Tecnativa to review?

File: base_global_discount/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    apply_global_discount = fields.Boolean(
+        string="Apply global discount",
+        default=True,
+        required=True,
```
Comment (pedrobaeza) [inline base_global_discount/models/product_product.py]: required in a boolean is a nonsense.

**Review by pedrobaeza — COMMENTED** [review]
I think in a business perspective more logical to use a negative flag: don't apply global discount, although the formula I prefer is to add a m2m in each global discount to exclude some products from them.

**Comment by ferran-S73** [comment]
@simahawk @pedrobaeza changes done, I haven't changed the field itself as to no disrupt our clients that are already using this feature

**Comment by pedrobaeza** [comment]
Well, you can put migration scripts, but keeping weird fields in the definitive version is not desirable.

**Comment by simahawk** [comment]
> Well, you can put migration scripts, but keeping weird fields in the definitive version is not desirable.

I agree. In this case you are allowed to change the version manually and add a pre/post migrate script to adapt existing installations.

**Comment by simahawk** [comment]
@Rferri44-S73 ping

**Comment by ferran-S73** [comment]
> @Rferri44-S73 ping

Hi @simahawk , I'm sorry I haven't been able to update this pr as requested. I have this in my todo list and will eventually apply the suggestions you and @pedrobaeza gave me. Sorry for the inconvenience

**Comment by cyrilmanuel** [comment]
Hi @Rferri44-S73  little ping have you the time to work on it  ? :)

**Comment by carolinafernandez-tecnativa** [comment]
Hi, could you please rebase? Thanks!

**Comment by dreispt** [comment]
This is blocking https://github.com/OCA/account-invoicing/pull/1338
What can be done to move forward?

**Review by carolinafernandez-tecnativa — APPROVED** [review]
LGTM, Thanks!

**Comment by carolinafernandez-tecnativa** [comment]
@simahawk can we proceed to merge? this blocks https://github.com/OCA/account-invoicing/pull/1338

File: base_global_discount/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Studio73 - Rafa Ferri <rafa.ferri@studio73.es>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
```
Comment (pedrobaeza) [inline base_global_discount/models/product_product.py]:
```suggestion
# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
```

File: base_global_discount/models/product_template.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Studio73 - Rafa Ferri <rafa.ferri@studio73.es>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
```
Comment (pedrobaeza) [inline base_global_discount/models/product_template.py]:
```suggestion
# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
```

**Comment by pedrobaeza** [comment]
Summarizing this one, apart from my latest comments, we are talking about not putting a flag in the product, that is very disrupting, but add a m2m in the global discounts for including or excluding. That way, you get even more granularity.

**Comment by ferran-S73** [comment]
Hi @pedrobaeza I've updated this PR adding a negative flag on the products as accorded on https://github.com/OCA/account-invoicing/pull/1725#issuecomment-2120503216

I have also updated th following PRs that depend on this one
- https://github.com/OCA/account-invoicing/pull/1725
- https://github.com/OCA/sale-workflow/pull/2343

Hope everything is alright now!

**Review by pedrobaeza — APPROVED** [review]
OK, I think it's good enough

/ocabot merge minor

**Pull request merged**